src/transactionRequests/handler.js

```diff
@@ -119,41 +117,69 @@ exports.getTransactionRequest = function (request, h) {
           'FSPIOP-Destination': request.headers['fspiop-source'],
           Date: new Date().toUTCString(),
           'FSPIOP-HTTP-Method': 'PUT',
-          'FSPIOP-URI': fspiopUriHeader
+          'FSPIOP-URI': `/transactionRequests/${request.payload.transactionRequestId}`
         },
         transformRequest: [(data, headers) => {
           delete headers.common.Accept
           return data
         }],
-        data: JSON.stringify(getTransactionRequestsResponse)
+        data: JSON.stringify(transactionRequestsResponse)
       }
-
       const res = await sendRequest(url, opts, request.span)
       Logger.info(`response: ${res.status}`)
-      if ((res.status !== Enums.Http.ReturnCodes.ACCEPTED.CODE) && (res.status !== Enums.Http.ReturnCodes.OK.CODE)) {
+      if (res.status !== Enums.Http.ReturnCodes.OK.CODE) {
         throw new Error(`Failed to send. Result: ${JSON.stringify(res)}`)
       }
     } catch (err) {
       Logger.error(err)
     }
   })()
 
-  return h.response().code(202)
+  return h.response().code(Enums.Http.ReturnCodes.ACCEPTED.CODE)
 }
 
-exports.callbackTransactionRequests = function (request, h) {
+exports.putTransactionRequest = function (request, h) {
+  Logger.info(`IN transactionRequests :: PUT /transactionRequests/${request.params.ID}, PAYLOAD: [${JSON.stringify(request.payload)}]`)
+  correlationCache.set(request.params.ID, request.payload)
```

Comment:
Would you not need to validate that the transaction ID is already there from the POST in the requestsCache?